### PR TITLE
Avatar search and thumbnails

### PR DESCRIPTION
**Pull request by netpro2k**
Adds a thumbnail owned file to avatar records and an "avatars" source to the search API. The avatars source currently only supports returning results by user. Also, the only user you are currently authorized to search for is the one you are authenticated as.

todo:
- [x] decide if its ok to get a listing of other people's avatars, restrict it if not
- [x] decide what to call the source, and if we want a "filter"
- [x] come up with a real placeholder image

### CODE REVIEW

File: lib/ret/avatar.ex [hunk 1]
```diff
@@ -93,6 +94,29 @@ defmodule Ret.Avatar do
     |> avatar_to_collapsed_files()
   end
 
+  def version(%Avatar{} = avatar) do
+    avatar.updated_at |> NaiveDateTime.to_erl |> :calendar.datetime_to_gregorian_seconds
+  end
+
+  def url(%Avatar{} = avatar) do
```
Comment (gfodor) [inline lib/ret/avatar.ex]: nit: you can re-write these to be one line:

`def url(%Avatar{} = avatar), do: "#{RetWeb.Endpoint.url()}/api/v1/avatars/#{avatar.avatar_sid}"`

File: lib/ret/media_search.ex [hunk 2]
```diff
@@ -216,7 +242,8 @@ defmodule Ret.MediaSearch do
         )
 
       res =
-        "https://api.twitch.tv/kraken/search/streams?#{query}" |> retry_get_until_success([{"Client-ID", client_id}])
+        "https://api.twitch.tv/kraken/search/streams?#{query}"
+        |> retry_get_until_success([{"Client-ID", client_id}])
```
Comment (brianpeiris) [inline lib/ret/media_search.ex]: Is your editor not respecting the 120 line length in the `.formatter.exs` file?

Comment (netpro2k) [reply lib/ret/media_search.ex]: ah must not be. will take a look as to why not

Comment (netpro2k) [reply lib/ret/media_search.ex]: Yep that was the issue

**Pull request merged**